_episodes/38.md

```diff
@@ -0,0 +1,475 @@
+underlying database is capable of.
+
+#### `GROUP_CONCAT()` in MySQL
+
+In a table where we have employees and their manager ID, but no direct way to
+list managers with all their employees, we can push down a query to MySQL and
+use `GROUP_CONCAT()` to combine them all into one column with this query:
+
+```
+SELECT
+  *
+FROM
+  TABLE(
+    mysql.system.query(
+      query => 'SELECT
+        manager_id, GROUP_CONCAT(employee_id)
+      FROM
+        company.employees
+      GROUP BY
+        manager_id'
+    )
+  );
+```
+
+#### MODEL clause in Oracle
+
+The `MODEL` clause in Oracle is an incredibly powerful way to manipulate and
+view data. As it's non-ANSI compliant, it's specific to Oracle, but if you want
+to use it, now you can! Through polymorphic table functions, you can generate
+and perform sophisticated calculations on multidimensional arrays - try saying
+that five times fast. We don't have the time to explain everything about how
+this feature works, but if you want clarification, you can check out
+[the Oracle documentation on MODEL](https://docs.oracle.com/cd/B19306_01/server.102/b14223/sqlmodel.htm)
+and try it out for yourself.
+
+```
+SELECT
+  SUBSTR(country, 1, 20) country,
+  SUBSTR(product, 1, 15) product,
+  year,
+  sales
+FROM
+  TABLE(
+    oracle.system.query(
+      query => 'SELECT
+        *
+      FROM
+        sales_view
+      MODEL
+        RETURN UPDATED ROWS
+        MAIN
+          simple_model
+        PARTITION BY
+          country
+        MEASURES
+          sales
+        RULES
+          (sales['Bounce', 2001] = 1000,
+          sales['Bounce', 2002] = sales['Bounce', 2001] + sales['Bounce', 2000],
+          sales['Y Box', 2002] = sales['Y Box', 2001])
+      ORDER BY
+        country'
+    )
+  );
+```
+
+Funnily enough, Oracle also supports polymorphic table functions, so if you
+wanted to, you could use the `query` function to then invoke a PTF in Oracle,
+including any of the hypothetical examples we went into above! PTFs inside of
+PTFs!
+
+## Question of the episode: Where are we at, and what's coming next?
+
+Right now, there's a few things on the radar for how to move forward with PTFs.
+The first and more simple task at hand is expanding the `query` function to
+other connectors. We started with the JDBC connectors, but we have also landed
+it for ElasticSearch and BigQuery, and there may still be more yet to come.
+
+On a broader scope, the reason this was the first PTF that was implemented is
+because Trino doesn't have to do anything to make it work. The next big step in
+powering PTFs up is to create an operator and make the engine aware of them, so
+that the engine can handle and process PTFs itself, which will open the door to
+the wide array of possibilities we explored earlier.
+
+And finally, once that's done, we plan on empowewring you, the Trino community,
+to go out and actually *make* some polymorphic table functions. Once the
```

Comment:
They can add them already. The limitations are: no table or descriptor arguments, and mandatory execution by the connector.